exporter/opentelemetry-exporter-prometheus/src/opentelemetry/exporter/prometheus/__init__.py

```diff
@@ -375,3 +378,18 @@ def _create_info_metric(
         info = InfoMetricFamily(name, description, labels=attributes)
         info.add_metric(labels=list(attributes.keys()), value=attributes)
         return info
+
+
+def _prometheus_metric_exporter_entrypoint() -> MetricReader:
+    """Implementation of opentelemetry_metrics_exporter entrypoint for prometheus exporter
+
+    This allows users to use the prometheus exporter with opentelemetry-instrument.
+    """
+
+    # Default values are specified in
+    # https://github.com/open-telemetry/opentelemetry-specification/blob/v1.24.0/specification/configuration/sdk-environment-variables.md#prometheus-exporter
+    start_http_server(
+        port=int(environ.get(OTEL_EXPORTER_PROMETHEUS_PORT, "9464")),
+        addr=environ.get(OTEL_EXPORTER_PROMETHEUS_HOST, "localhost"),
```

Comment:
The Prometheus exporter doesn't already handle these environment variables because it leave the responsibility of starting the server up to the user. I think this is for flexibility, for example so the user could [server the prometheus endpoint in an existing flask app](https://github.com/prometheus/client_python#flask).

Do you think this should be moved into the `PrometheusExporter` constructor or is it good here?

Reply:
> Do you think this should be moved into the PrometheusMetricReader constructor or is it good here?

I'm confused as to how moving the env/port configuration into `PrometheusMetricReader` will address the issue of flexibility? Isn't the question whether or not we want to let OpenTelemetry Prometheus exporter automatically the server instead of the current behavior of leaving the user start up the server? As from [this question](https://github.com/open-telemetry/opentelemetry-python/pull/3413#discussion_r1336373678), should we be consistent between manual and auto instrumentation?

Reply:
> I'm confused as to how moving the env/port configuration into `PrometheusMetricReader` will address the issue of flexibility?

It doesn't it, would make things less flexible. The problem is the exporter should respect these envvars but if it's not starting the server it can't really do that. Here's what other languages do:

- Go exporter does similar to what we do. You have to set up the prometheus client handler, [for example](https://github.com/open-telemetry/opentelemetry-go/blob/0f09b9b2d8654c05ed35f8b4065c5eddf5f0211c/example/prometheus/main.go#L93-L93)
- JS exporter starts the http server automatically, respecting the passed host/port, envvars, or uses the default. [Code](https://github.com/open-telemetry/opentelemetry-js/blob/main/experimental/packages/opentelemetry-exporter-prometheus/src/PrometheusExporter.ts). Note they are not wrapping any Prometheus client lib
- Java exporter does similar to JS. It starts its own HTTP server respecting the host/port. [Code](https://github.com/open-telemetry/opentelemetry-java/blob/main/exporters/prometheus/src/main/java/io/opentelemetry/exporter/prometheus/PrometheusHttpServer.java#L95).

Maybe we should consider exposing `_AutoPrometheusMetricReader` to users so they can choose either option. Go will need a similar option if they want to respect the specified defaults and environment variables.

Reply:
Feel free to bring up this topic in the SIG. Regardless, I don't think it's a blocking issue.

Reply:
👍 I haven't made `_AutoPrometheusMetricReader` public or changed the existing `PrometheusMetricReader` at all here